File: Books.Web/Scripts/ViewModels/AuthorViewModel.js
var AuthorViewModel = function(author) {
    var self = this;

    self.firstName = ko.observable(author.firstName);
    self.firstName.extend({
        required: true,
        maxLength: 20
    });

    self.lastName = ko.observable(author.lastName);
    self.lastName.extend({
        required: true,
        maxLength: 20
    });

    self.update = function (updAuthor) {
        if (updAuthor == null) return;

        self.firstName(updAuthor.firstName);
        self.lastName(updAuthor.lastName);
    }
}

File: Books.WebApi/ViewModels/AuthorViewModel.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Books.WebApi.ViewModels
{
	public class AuthorViewModel
	{
		/// <summary>
		/// Author ID
		/// </summary>
		public Guid Id { get; set; }

		/// <summary>
		/// First name
		/// </summary>
		[Required]
		[MaxLength(20)]
		public string FirstName { get; set; }

		/// <summary>
		/// Last name
		/// </summary>
		[Required]
		[MaxLength(20)]
		public string LastName { get; set; }

		/// <summary>
		/// Book ID
		/// </summary>
		public Guid BookId { get; set; }
	}
}
